addons/mail/static/src/components/selectable_partners_list/selectable_partners_list.js

```diff
@@ -0,0 +1,120 @@
+
+
+    get partners() {
+        const partners = this.env.messaging.selectablePartnersList.selectablePartners.map(partner => {
+            return {
+                avatarUrl: partner.avatarUrl,
+                id: partner.id,
+                isInitialMember: this.props.initialMembers.includes(partner.id),
+                isSelected: this.selectedPartners.includes(partner.id),
+                name: this._getPartnerName(partner.nameOrDisplayName),
+            }
+        });
+        return partners;
+    }
+
+    get thread() {
+        return this.env.models['mail.thread'].get(this.props.threadLocalId);
+    }
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _close() {
+        this.trigger('o-popover-close');
+    }
+
+    /**
+     * @private
+     * @param {string} partnerName
+     *
+     * @returns {string}
+     */
+    _getPartnerName(partnerName) {
+        const commaIndex = partnerName.lastIndexOf(",");
+        if (commaIndex !== -1) {
+            return partnerName.substring(commaIndex + 2, partnerName.length);
+        }
+        return partnerName;
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    async _onClickCreateGroupChat() {
+        if (this.props.initialMembers.length > 2 && this.props.initialMembers.length !== this.selectedPartners.length) {
+            await this.env.services.rpc(({
+                model: 'mail.channel',
+                method: 'channel_invite',
+                args: [this.thread.id],
+                kwargs: {
+                    partner_ids: this.selectedPartners,
+                },
+            }));
+            // Trigger rendu de la topbar
+        } else if (this.props.initialMembers.length !== this.selectedPartners.length) {
+            this.env.messaging.openGroupChat(this.selectedPartners);
+        }
+
+        this._close();
+    }
+
+    _onSelectedPartner(ev) {
+        const selected = this.selectedPartners.includes(ev.detail.id);
+        if (ev.detail.selected && !selected) {
+            this.selectedPartners.push(ev.detail.id);
+            return;
+        }
+
+        if (!ev.detail.selected && selected) {
+            this.selectedPartners = this.selectedPartners.filter(partnerId => partnerId !== ev.detail.id);
+        }
+    }
+}
+
+Object.assign(SelectablePartnersList, {
+    components,
+    defaultProps: {
+        inputSearch: "",
+    },
+    props: {
+        initialMembers: {
+            type: Array,
+        },
+        inputSearch: {
+            type: String,
+        },
+        threadLocalId: {
+            type: String,
+        }
+    },
+    template: 'mail.SelectablePartnersList',
+});
```

Comment:
At first glance it looks like there is too much logic in this component that should be moved to a corresponding model instead.